elastic/kibana
# [Discover] Fix filtering out custom meta fields of Elasticsearch plugins enhanced documents

## Summary

In #114517 multiple flattening of documents implementations were unified. This caused a bug with meta fields. The implementation assumed meta fields would match a given set of meta fields, filtering out all other fields. However, there's a way to extend the set of meta fields in Kibana with Elasticsearch plugins like the [Mapper Size Plugin](https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-size.html)
There was not way to view the `_size` meta field in Discover (apart from the JSON view). This fix changes the implementation to exclude the fields that should not be merged ('_source', '_type')

### Testing
* Install the Mapper Size Plugin to test this. 
https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-size.html
* You need to install a previously release ES version, since it doesn't work with snapshots
   ` yarn es archive https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-8.3.2-darwin-aarch64.tar.gz`
* Go to new downloaded archive `cd .es/elasticsearch-8.3.2-darwin-aarch64`
* There you can install the plugin by `./bin/elasticsearch-plugin install mapper-size`
* You need to disable security by setting`xpack.security.enabled: false` in `./config/elasticsearch.yml`
* Then you can start Elasticsearch by executing` ES_PATH_CONF=./config ./bin/elasticsearch` in `es/elasticsearch-8.3.2-darwin-aarch64`
* You need to configure Kibana to work with this previous version by setting `elasticsearch.ignoreVersionMismatch: true
` in your `config/kibana.dev.yml`
* Then you should be able to start Kibana
* Add `_size` to `metaFields` of **Advanced settings**
* Configure mapping to use `_size` and ingest data like it is described in  https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-size-usage.html
* Go to Discover, create a data view for the new index with the ingested data
*  `_size` should be displayed with a value when expanding a document, and you can add the field to Discover's document explorer


Fixes #121394

### Checklist
